hudi-spark-datasource/hudi-spark-common/src/main/scala/org/apache/hudi/BaseFileOnlyRelation.scala

```diff
@@ -160,9 +160,6 @@ class BaseFileOnlyRelation(sqlContext: SQLContext,
         fileFormat = fileFormat,
         optParams)(sparkSession)
     } else {
-      val readPathsStr = optParams.get(DataSourceReadOptions.READ_PATHS.key)
```

Comment:
Using `READ_PATHS` may miss glob paths if user use `spark.read.format("hudi").load("glob.path")`, so change here to directly use `globPaths`(which cover both)